gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/NotifierServiceImplTest.java

```diff
@@ -256,5 +258,33 @@ void should_trigger_email_to_additional_opted_in_recipients() {
             verify(emailNotifierService).trigger(eq(executionContext), eq(ApiHook.API_STARTED), eq(params), recipientsCaptor.capture());
             assertThat(recipientsCaptor.getValue()).containsExactlyInAnyOrder("additional@gio.test", "recipient@gio.test");
         }
+
+        @Test
+        @SneakyThrows
+        void should_not_trigger_notification_if_hook_not_present() {
+            final Map<String, Object> params = Map.of();
+            final ExecutionContext executionContext = new ExecutionContext();
+
+            when(
+                genericNotificationConfigRepository.findByReferenceAndHook(
+                    ApiHook.API_STARTED.name(),
+                    NotificationReferenceType.API,
+                    "api-id"
+                )
+            )
+                .thenReturn(Collections.emptyList());
+
+            cut.triggerGenericNotifications(
+                executionContext,
+                ApiHook.API_STARTED,
+                NotificationReferenceType.API,
+                "api-id",
+                params,
+                List.of()
+            );
+
+            verify(emailNotifierService, times(0)).trigger(any(), any(), any(), any());
```

Comment:
I may be wrong but you can replace the `verify` by a `verifyNoInteractions` no ?